torch/nn/utils/weight_norm.py

```diff
@@ -36,7 +36,10 @@ def apply(module, name: str, dim: int) -> 'WeightNorm':
         fn = WeightNorm(name, dim)
 
         weight = getattr(module, name)
-
+        if isinstance(weight, UninitializedParameter):
```

Comment:
I'm not in love with these norms being aware of uninitialized parameters. Would you it be possible to have the lazy linear modules throw this error on their own? 

For example, the lazy linear module could define two empty tensors, _bias and _weight, and then when bias or weight is requested the module itself could throw an error like this, I think?

Reply:
Sorry, right now I don't understand how adding two extra parameters can help to detect these kinds of errors.

Reply:
I'm probably not being very clear about this. Maybe we should leave it for future work.

Reply:
I understood it from your example in another comment. However, this affects the current design so we should discuss this better. I will reply in that comment.

Reply:
Sounds good. Let's leave this as-is for now.